src/main/java/edu/cornell/kfs/fp/batch/service/impl/CorporateBilledCorporatePaidRouteStepReportServiceImpl.java

```diff
@@ -0,0 +1,136 @@
+        reportWriterService.writeFormattedMessageLine(MessageFormat.format(suummaryDetailFormat, "in saved status", totalCbcpCount));
+        reportWriterService.writeFormattedMessageLine(MessageFormat.format(suummaryDetailFormat, "routed", totalRoutedCount));
+        reportWriterService.writeFormattedMessageLine(MessageFormat.format(suummaryDetailFormat, "failed to route", totalErrorCount));
+        reportWriterService.writeNewLines(3);
+    }
+
+    protected void buildErrorSection(Map<String, String> documentErrors) {
+        reportWriterService.writeSubTitle(configurationService.getPropertyValueAsString(
+                CuFPKeyConstants.MESSAGE_CBCP_ROUTE_STEP_REPORT_ERROR_SUBHEADER));
+        if (documentErrors.size() > 0) {
+            String errorDetailFormat = configurationService.getPropertyValueAsString(CuFPKeyConstants.MESSAGE_CBCP_ROUTE_STEP_REPORT_ERROR_DETAIL_LINE);
+            for (String documentNumber : documentErrors.keySet()) {
+                String errorMessage = documentErrors.get(documentNumber);
+                LOG.info("buildErrorSection, document " + documentNumber + " failed to route due to: " + errorMessage);
+                reportWriterService.writeFormattedMessageLine(MessageFormat.format(errorDetailFormat, documentNumber, errorMessage));
+            } 
+        } else {
+            reportWriterService.writeNewLines(1);
+            reportWriterService.writeSubTitle(configurationService.getPropertyValueAsString(
+                    CuFPKeyConstants.MESSAGE_CBCP_ROUTE_STEP_REPORT_NO_ERRORS));
+        }
+    }
+    
+    @Override
+    public String buildValidationErrorMessage(ValidationException validationException) {
+        try {
+            Map<String, AutoPopulatingList<ErrorMessage>> errorMessages = GlobalVariables.getMessageMap().getErrorMessages();
+            return errorMessages.values().stream().flatMap(List::stream)
+                    .map(this::buildValidationErrorMessageForSingleError)
+                    .collect(Collectors.joining(KFSConstants.NEWLINE,
+                            validationException.getMessage() + KFSConstants.NEWLINE, KFSConstants.NEWLINE));
+        } catch (RuntimeException e) {
+            LOG.error("buildValidationErrorMessage: Could not build validation error message", e);
+            return CuFPConstants.ALTERNATE_BASE_VALIDATION_ERROR_MESSAGE;
+        }
+    }
+
+    protected String buildValidationErrorMessageForSingleError(ErrorMessage errorMessage) {
+        String errorMessageString = configurationService.getPropertyValueAsString(errorMessage.getErrorKey());
+        if (StringUtils.isBlank(errorMessageString)) {
+            throw new RuntimeException("Cannot find error message for key: " + errorMessage.getErrorKey());
+        }
+        
+        Object[] messageParameters = (Object[]) errorMessage.getMessageParameters();
```

Comment:
Is the cast to Object[] from String[] needed? 
Also, is using "!ObjectUtils.IsNull(messageParameters)" preferred over "!= null" for an array of objects?

Reply:
I reused this code from another part of KFS.  I think using CollectionUtils.isNotEmpty would be best here.  I shall switch it.  Thanks!

Reply:
I reused this code from another part of KFS.  I think using CollectionUtils.isNotEmpty would be best here.  I shall switch it.  Thanks, I shall clean  it up